multithreading/consumers.py

```diff
@@ -29,6 +28,23 @@ class DataConsumer(Consumer):
     def __init__(self, lock, queue, gui):
         super().__init__(lock, queue)
         self.gui = gui
+        # TODO when we implement procedures
+        self.shutdown_procedure = None
+
+    def requires_shutdown(self, message_id):
+        """
+        Determines if the message is severe enough to trigger the shutdown procedure.
+        """
+        return get_message_severity(message_id) <= CRITICAL_ERROR
```

Comment:
the higher the message id, the more critical it is?
return message_id >= critical 

this would make more sense, but you're the one that knows the message id scale meaning

Reply:
same question